File: src/utils/terminal.rs
use std::io::Write;
use rpassword;



pub fn show_menu(title: &str, items: &[&str], main_menu: bool) -> u32 {
  let full_title: String = String::from("Interative Rust CMD :: ") + title;
  
  println!("{}", full_title);
  println!("{}", "=".repeat(full_title.len()));

  display_options(items, main_menu);
  display_prompt();

  let mut choice: String = String::new();

  std::io::stdin()
    .read_line(&mut choice)
    .unwrap();

  println!("");

  match choice.trim().parse() {
    Ok(num) => num,
    Err(..) => 0
  }
}


pub fn display_prompt() {
  print!("\nEscolha a opção: ");
  std::io::stdout().flush().unwrap();
}


pub fn display_options(items: &[&str], main_menu: bool) {
  for (i, item) in items.iter().enumerate() {
    println!("{} - {}", i + 1, item);
  }
  println!("{}", if main_menu {"* - Exit"} else {"* - Back"});
}


pub fn clear_screen() {
  println!("{esc}c", esc=27 as char);
}


pub fn wait_any() {
  rpassword::prompt_password("Press any key to continue...").unwrap();
}

File: src/utils/terminal.cpp
#include <iomanip>
#include <iostream>
#include <string>
#include <vector>


#include "terminal.h"


namespace terminal {


void clear_screen() {
  #if defined _WIN32
    system("cls");
  #elif defined(__LINUX__) || defined(__gnu_linux__) || defined(__linux__) || defined(__APPLE__)
    system("clear");
  #endif
}


void wait_any_key() {
  std::cout << "\n";
  system("PAUSE");
}


unsigned int display_prompt(
  const std::string prompt,
  unsigned int max_attempts
) {

  unsigned int input;
  unsigned int attpt;

  while (true) {

    std::cout << "\n" << prompt << "> ";
    std::cin >> input;

    if (!std::cin.fail() || attpt == max_attempts) {
      break;
    } else {
      std::cin.clear();
      std::cin.ignore(65535, '\n');
    }

    attpt++;
  }

  return input;
}

void display_options(
  const std::vector<std::string> &items, 
  bool main_menu
) {

  unsigned short int i;
  
  for (i = 0; i < items.size(); i++) {
    std::cout << (i + 1) << " - " << items[i] << "\n";
  }

  if (main_menu) {
    std::cout << "0 - Exit\n";
  }
  else {
    std::cout << "0 - Back\n";
  }
}

unsigned int show_menu(const std::string title,
                       const std::vector<std::string> &items, bool main_menu) {

  const std::string full_title = "Interative C++ CMD :: " + title;

  std::cout << full_title << "\n";
  std::cout << std::setw(full_title.size() + 2) << std::setfill('=') << "\n\n";

  display_options(items, main_menu);

  return display_prompt("Choose an option", 3);
}
} // namespace terminal
